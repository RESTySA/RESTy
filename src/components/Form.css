.Buttons{
    display:flex;
    flex-direction: row;
    justify-content:space-between;

    input{
        display:none;
    }

    label{
        padding: 4px 15px;
        margin: 10px 0;
        font-weight: 550;
    }

    label:hover{
        background-color:rgb(217, 230, 235);
    }

    button{
        height:20px;
        align-self: center;
        background-color: white;
        border-radius:5px;
    }

    input:checked + label{
        background-color:#2d5986;
        color: white;
    }
}

form{
    margin: 20px 0 0 0;
}