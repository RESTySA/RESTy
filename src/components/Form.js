import React, { Component } from 'react';
import PropTypes from 'prop-types';


export default class Form extends Component {

    static propTypes = {
      url: PropTypes.string.isRequired,
      handleRouteChange: PropTypes.func.isRequired,
      handleSubmit: PropTypes.func.isRequired,
      handleUserInput: PropTypes.func.isRequired,
      handleURLInput: PropTypes.func.isRequired
    }

    render() {
      const { handleRouteChange, handleSubmit, handleUserInput, handleURLInput } = this.props;
      return (
        <form onSubmit={handleSubmit}>
          <input type="string" placeholder="URL" onChange={handleURLInput} />
          <input onChange={handleRouteChange} type="radio" name="method" value="get" /> <span>GET</span>
          <input onChange={handleRouteChange} type="radio" name="method" value="post" /> <span>POST</span>
          <input onChange={handleRouteChange} type="radio" name="method" value="put" /> <span>PUT</span>
          <input onChange={handleRouteChange} type="radio" name="method" value="delete" /> <span>DELETE</span>
          <textarea onChange={handleUserInput} placeholder="Raw JSON Body" rows={10} cols={70} />
          <button onClick={handleSubmit}>Go!</button>
        </form>
      );
    }

}
