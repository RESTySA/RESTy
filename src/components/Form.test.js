import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  const test = {
    url: 'test.com',
    handleChange: (() => {}),
    handleSubmit: (() => {})
  };

  it('renders Form', () => {
    const wrapper = shallow(<Form 
      url={test.url}
      handleChange={test.handleChange}
      handleSubmit={test.handleSubmit} />);
    expect(wrapper).toMatchSnapshot();
  });
});