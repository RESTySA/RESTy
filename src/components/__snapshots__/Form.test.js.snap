// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="url"
    onChange={[Function]}
    placeholder="URL"
    type="string"
  />
  <div
    className="Buttons"
  >
    <input
      id="get"
      name="method"
      onChange={[Function]}
      type="radio"
      value="get"
    />
    <label
      htmlFor="get"
    >
      GET
    </label>
    <input
      id="post"
      name="method"
      onChange={[Function]}
      type="radio"
      value="post"
    />
    <label
      htmlFor="post"
    >
      POST
    </label>
    <input
      id="put"
      name="method"
      onChange={[Function]}
      type="radio"
      value="put"
    />
    <label
      htmlFor="put"
    >
      PUT
    </label>
    <input
      id="delete"
      name="method"
      onChange={[Function]}
      type="radio"
      value="delete"
    />
    <label
      htmlFor="delete"
    >
      DELETE
    </label>
    <button
      onClick={[Function]}
    >
      Go!
    </button>
  </div>
  <textarea
    cols={70}
    name="userInput"
    onChange={[Function]}
    placeholder="Raw JSON Body"
    rows={10}
  />
</form>
`;
